In [1]:
import numpy as np
from scipy import stats

# Домашнее задание 7   

## Задача 1

> Даны две  независимые выборки. Не соблюдается условие нормальности  
> x  380,420, 290  
> y 140,360,200,900  
> Сделайте вывод по результатам, полученным с помощью функции

$\color{red}Критерий\space Манна – Уитни$

$H_0:\mu_0=\mu_1$  
$H_1:\mu_0\not=\mu_1$

In [2]:
x = np.array([380, 420, 290])
y = np.array([140, 360, 200, 900])
stats.mannwhitneyu(x, y)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

$\color{blue}Ответ:$ p-value = 0.628, что дает нам право сделать выбор в пользу нулевой гипотезы, т.к. даже при α=0.1 p-value > α.

## Задача 2

> Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?  
> 1е измерение до приема препарата: 150, 160, 165, 145, 155  
> 2е измерение через 10 минут: 140, 155, 150,  130, 135  
> 3е измерение через 30 минут: 130, 130, 120, 130, 125  

$\color{red}Критерия\space Фридмана$

$H_0:\mu_0=\mu_1$  
$H_1:\mu_0\not=\mu_1$  
$\alpha=0.05$

In [3]:
a = ([150, 160, 165, 145, 155])
b = ([140, 155, 150, 130, 135])
c = ([130, 130, 120, 130, 125])

stats.friedmanchisquare(a, b, c)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

$\color{blue}Ответ:$ pvalue = 0.008 < $\alpha = 0.05$ Статистические различия есть

## Задача 3

> Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.  

$\color{red}Критерий\space Уилкоксона  $

$H_0:\mu_0=\mu_1$  
$H_1:\mu_0\not=\mu_1$  
$\alpha=0.05$

In [4]:
stats.wilcoxon(a, b)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

$\color{blue}Ответ:$ pvalue = 0.06 > $\alpha = 0.05$ Статистических различий нет

## Задача 4

> Даны 3 группы  учеников плавания.  
> В 1 группе время на дистанцию 50 м составляют:  
> 56, 60, 62, 55, 71, 67, 59, 58, 64, 67  
> Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53  
> Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54  

$\color{red}Критерий\space Крускала –Уоллиса  $

$H_0:\mu_0=\mu_1$  
$H_1:\mu_0\not=\mu_1$  
$\alpha=0.05$

In [5]:
a = ([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
b = ([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
c = ([57, 67, 49, 48, 47, 55, 66, 51, 54])
stats.kruskal(a, b, c)

KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

$\color{blue}Ответ:$ pvalue = 0.065 > $\alpha = 0.05$ Статистических различий нет

## Задача 5

> Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%  
> 
> 2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

$H0: M = 2.5$

$H1: M \not= 2.5$

$\sigma =\frac{1}{n-1}\sum(X_i-\overline{X})^2$

$t=\frac{\overline{X}-\mu}{\frac{\sigma_н}{\sqrt{n}}}$

In [6]:
X = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
mean_X = X.mean()
std_X = X.std(ddof=1)
t_fact = (mean_X - 2.5) / std_X * np.sqrt(10)
t_fact

0.563061366180296

По таблице критических точек распределения Стьюдента для двухсторонней области найдем критическую точку по
уровню значимости α = 0,05 и числу степеней свободы k = 9 

In [7]:
stats.t.ppf(1 - 0.05 / 2, 9)

2.2621571627409915

$t_{crit}$ = 2.26

$t_{fact}$ = 0.563

$|t_{fact}|<t_{crit}$

$\color{blue}Ответ:$ Партия изготавливается со средним арифметическим 2,5 см.